### ISSUE TITLE
Message routing by interfaces

### ISSUE THREAD

**Issue by alexander-yevsyukov** [SpineEventEngine/core-java#987]
Currently the routing schemas uses concrete classes of messages. See the method `MessageRouting.apply(M,C)`.

This requires repeating the same routing path all over again. 

Here's the code from the example application which shows the current usage:
```java
 eventRouting()
    .route(CardCreated.class, (event, context) -> withId(event.getBoardId()))
    .route(ColumnCreated.class, (event, context) -> withId(event.getBoardId()));
```
Instead, we can make the event classes implement the common interface `BoardEvent` with the `getBoardId()` method, and then have more compact form:

```java
eventRouting().route(BoardEvent.class, (event, context) -> withId(event.getBoardId()));
```

Such simplification of routing will be helpful when there are many events that are routed by ID of another aggregate, and which is not in the first field of a routed message.

**Comment by alexander-yevsyukov** [comment]
Here's a piece from `MirrorRepository` which can also benefit from this feature:

```java
    private void prepareRouting() {
        eventRouting()
                .route(EntityStateChanged.class,
                       (message, context) -> targetsFrom(message.getId()))
                .route(EntityArchived.class,
                       (message, context) -> targetsFrom(message.getId()))
                .route(EntityDeleted.class,
                       (message, context) -> targetsFrom(message.getId()))
                .route(EntityExtractedFromArchive.class,
                       (message, context) -> targetsFrom(message.getId()))
                .route(EntityRestored.class,
                       (message, context) -> targetsFrom(message.getId()));
    }
```

**Comment by alexander-yevsyukov** [comment]
Addressed in #1038.
